pandas/core/internals/managers.py

```diff
@@ -2405,7 +2406,7 @@ def _preprocess_slice_or_indexer(
 def make_na_array(dtype: DtypeObj, shape: Shape, fill_value) -> ArrayLike:
     if isinstance(dtype, DatetimeTZDtype):
         # NB: exclude e.g. pyarrow[dt64tz] dtypes
-        i8values = np.full(shape, fill_value._value)
+        i8values = np.full(shape, Timestamp(fill_value)._value)
```

Comment:
i think this may mess up if your dtype has a different unit than the fill_value.  can you add a test where this is the case?  also needs a test for the original bug

Reply:
> also needs a test for the original bug

My bad! I forgot to add that. Now added that👍
Thanks for the reminding.

Reply:
> i think this may mess up if your dtype has a different unit than the fill_value. can you add a test where this is the case?

Are you talking about something like this? where the value will be timestamp but its dtype will be different?
```py
df = pd.DataFrame([{'A':None}])
df_2 = pd.DataFrame([{'A': pd.to_datetime("1990-12-20 00:00:00+00:00")}], dtype="int32")
cd = pd.concat([df, df_2])
```

sorry! if I did not get it correctly. Could you please explain it in detail?